src/internal/isCoercible.js

```diff
@@ -0,0 +1,10 @@
+import { both, complement, either, hasIn, __ } from 'ramda';
+
+import { isObj, isSymbol, neither } from '..';
+
+const isCoercible = neither(
+  isSymbol,
+  both(isObj, complement(either(hasIn('toString', __), hasIn('valueOf', __))))
```

Comment:
Here you don't need `__`.

```js
both(isObj, complement(either(hasIn('toString'), hasIn('valueOf'))))
```

Usually, you will need it when you need to fill one parameter in the middle of a ramda function.

```js
// RamdaJs example
const greet = R.replace('{name}', R.__, 'Hello, {name}!');
greet('Alice'); //=> 'Hello, Alice!'
```

You can call
```js
const greet = R.replace('{name}', 'Alice');
greet('Hello, {name}!'); //=> 'Hello, Alice!'
```

Because, since the last parameter was not given, the `greet` function will wait for it.
Back to the function `const greet = R.replace('{name}', R.__, 'Hello, {name}!');`

Here, the `greet` function it's like:
```js
const greet = name => R.replace('{name}', name, 'Hello, {name}!');
```

So, with `__` you can choose with the parameter you want to pass later 😄

Reply:
Nice catch, thank you